test/test_profiler.py

```diff
@@ -138,6 +140,182 @@ def test_json_output():
+    assert "CPU profile" in output["name"]
+
+    assert "frames" in output["shared"]
+    speedscope_frame_list = output["shared"]["frames"]
+
+    # Distinct functions called stores indices in key-value pairs because function
+    # index lookup needed for Speedscope event list tests. Were we not testing
+    # the event list, the distinct functions called could be stored as a tuple.
+    distinct_functions_called = {
+        "test_speedscope_output": 0,
+        "long_function_a": 1,
+        "sleep": 2,
+        "long_function_b": 3,
+    }
+    assert len(speedscope_frame_list) == len(distinct_functions_called)
+    speedscope_frame_fields = tuple([field.name for field in dataclasses.fields(SpeedscopeFrame)])
+    for (function_name, frame_index) in distinct_functions_called.items():
+        for frame_field in speedscope_frame_fields:
+            assert frame_field in speedscope_frame_list[frame_index], (
+                f"Field named '{frame_field}' not in Speedscope output"
+                f"at 'shared.frames[{frame_index}]'"
+            )
+        frame_index_name = speedscope_frame_list[frame_index]["name"]
+        assert frame_index_name == function_name, (
+            f"Speedscope output 'shared.frames[{frame_index}].name' has value "
+            f"{frame_index_name} != {function_name}"
+        )
+
+    speedscope_profile_list = output["profiles"]
+    assert len(speedscope_profile_list) == 1
+    speedscope_profile = speedscope_profile_list[0]
+    speedscope_profile_fields = ("type", "name", "unit", "startValue", "endValue", "events")
+    for profile_field in speedscope_profile_fields:
+        assert (
+            profile_field in speedscope_profile
+        ), f"Field named '{profile_field}' not in Speedscope output at 'profiles[0]'"
+
+    # speedscope_profile["endValue"] is not tested because a fake_time mock
+    # timer is used to replace time.time, and this mock causes session.duration
+    # to differ from self._event_time just before exiting SpeedscopeRenderer.render
+    start_time_in_seconds = 0.0
+    event_time_in_seconds = start_time_in_seconds
+    assert speedscope_profile["type"] == "evented"
+    assert speedscope_profile["unit"] == "seconds"
+    assert speedscope_profile["startValue"] == start_time_in_seconds
+
+    """Two helper functions for constructing a timeline and testing it"""
+
+    def update_event_time_on_sleep_close(time_increment_in_seconds: float):
+        """Updates running total of event time in seconds and returns updated value"""
+        nonlocal event_time_in_seconds
+        event_time_in_seconds += time_increment_in_seconds
+        return event_time_in_seconds
+
+    def get_approx_abs_tol(time_in_seconds: float):
+        """Computes absolute tolerance for given time value to account for CI time jitter"""
+        uncertainty_per_quarter_second = 0.1
+        absolute_tolerance = (time_in_seconds / 0.25) * uncertainty_per_quarter_second
+        return absolute_tolerance
```

Comment:
We can use pytest.approx(rel=...) for this, instead of the function call. It would be `uncertainty_per_second = 0.4`... which seems very large! Are you sure we need this? I'd have thought that the fake_time would make CI jitter irrelevant.

Reply:
Maybe try doing pytest.approx() without setting abs or rel, I think it might work...

Reply:
Those comparisons don't require tolerances on my machine. I pattern-matched the tolerances to the following section of code, which motivated the 0.1s absolute tolerance per 0.25s of fake time:
https://github.com/joerick/pyinstrument/blob/c2e98aaca9581dd9a25373c51b704bf45fce62e8/test/test_profiler.py#L105-L108.

I'll remove the tolerances. All of the floating point numbers involved should be exactly representable per the IEEE-754 standard, so as long as the fake times aren't affected by external factors, tolerances should be unnecessary.

Reply:
Tolerances and local methods removed in [`97925fb2`](https://github.com/joerick/pyinstrument/pull/160/commits/97925fb284a34a111b7b99bbef8cd0bb72e7c584).

Reply:
Ah, gotcha, thanks for the explanation! Those tolerances look like a mistake, a hangover from the time before I had fake_time implemented. Sorry they misled you!